src/actors/common/cache.rs

```diff
@@ -325,13 +348,15 @@ impl<T: CacheItemRequest> Cacher<T> {
             }
         };
 
-        let future = channel
-            .map_err(move |_cancelled_error| {
-                let message = format!("{} computation channel dropped", name);
-                Arc::new(io::Error::new(io::ErrorKind::Interrupted, message).into())
-            })
-            .and_then(|shared| (*shared).clone());
+        let future = channel.unwrap_or_else(move |_cancelled_error| {
+            let message = format!("{} computation channel dropped", name);
+            Err(Arc::new(
+                io::Error::new(io::ErrorKind::Interrupted, message).into(),
+            ))
+        });
+        // TODO(flub): what was this cloning??
+        // .and_then(|shared| (*shared).clone());
```

Comment:
This is a [shared](https://docs.rs/futures/0.1.30/futures/future/trait.Future.html#method.shared) receiver of a oneshot channel, and it used to clone the inner data. In futures 0.3, this has changed, and the data has to be clonable to begin with.

Reply:
thanks